lib/puppet/util/run_mode.rb

```diff
@@ -85,6 +85,50 @@ def run_dir
       def log_dir
         which_dir("/var/log/puppetlabs/puppet", "~/.puppetlabs/var/log")
       end
+
+      def pkg_config_path
+        '/opt/puppetlabs/puppet/lib/pkgconfig'
+      end
+
+      def gem_cmd
+        '/opt/puppetlabs/puppet/bin/gem'
+      end
+    end
+
+    class LinuxFHSRunMode < RunMode
+      def conf_dir
+        which_dir("/etc/puppet", "~/.puppetlabs/etc/puppet")
+      end
+
+      def code_dir
+        which_dir("/etc/puppet/code", "~/.puppetlabs/etc/code")
+      end
+
+      def var_dir
+        which_dir("/var/lib/puppet", "~/.puppetlabs/opt/puppet/cache")
+      end
+
+      def public_dir
+        which_dir("/var/lib/puppet/public", "~/.puppetlabs/opt/puppet/public")
+      end
+
+      def run_dir
+        which_dir("/run/puppet", "~/.puppetlabs/var/run")
+      end
+
+      def log_dir
+        which_dir("/var/log/puppet", "~/.puppetlabs/var/log")
+      end
+
+      def pkg_config_path
+        # TODO: not always
+        '/usr/lib64/pkgconfig'
```

Comment:
Should it return `nil` instead since we don't really want to override it?

Reply:
I think so, as the default pkgconfig path should be picked up automatically without the need for us to specify it.